### DISCUSSION TITLE
Installation without Composer

### DISCUSSION THREAD

**Discussion by jparker1986** [OpenMage/magento-lts#2022] [Questions & Answers] [answered]
Hi Guys,

New to the thread and looking to know if anyone has documentation to install this without a composer? Not sure if the composer would break my website or not. 

Great to see the project continuing for M1, keep up the good work.

**Comment by justinbeaty** [comment]
@jparker1986 You can clone the git repo or download a zip file from: https://github.com/OpenMage/magento-lts/releases

**Reply by jparker1986** [reply]
Thank you,

I seen on the website that you can run a command line curl -fsSL https://migrate.openmage.org | sh

Does this do the same thing?

**Reply by justinbeaty** [reply]
I've never used the migration script personally, so I don't know exactly how it works. I would definitely make a backup of the website and ideally try this on a local copy first.

Theoretically you could also just extract the current OpenMage zip on top of your current install, but I don't know if your current store has ever had core files modified.

**Reply by jparker1986** [reply]
I am not sure if we have modified core files or not but when applying the usual Magento SUPEE patches I would assume if they were modified then they would have been overwritten.

Our latest version of Magento was 1.9.4.1, I am not sure if that is the latest version or not but will have a go at replacing it and restoring a backup if it goes wrong.

**Comment by jparker1986** [comment]
Hi Guys

Attempted the migration today and got the following error:

git --git-dir=/dev/null apply --ignore-whitespace --check var/magento-ce-1.9.4.1-openmage-lts-v19.4.8.patch

Any idea?

**Comment by jparker1986** [comment]
Guys managed to sucessfully migrate to the service but the Migration installs to OpenMage LTS 19.4.8.

After Migration do I need to install the updates manually or via Git? Any newbie guide on how to do this would be mostly welcome.

**Reply by justinbeaty** [reply]
Looking at the migration script, it doesn't look like it clones the repo, but can you check `git status` in the root of the magento dir?

**Comment by jparker1986** [comment]
We have our own staging environment on git, so are we saying we need to
migrate then git clone?

**Reply by justinbeaty** [reply]
What I'm getting at, is if the root of your Magento-LTS install is in fact a git repo, you should be able to just do `git pull`. It's not quite clear to me the differences between your staging and prod environments.

**Comment by jparker1986** [comment]
Sorry, the website as a whole is part of a git repo. So updates to our
website would be done via a staging server, changes pushed to git and
pulled to live.

Presently the openmage lts is on our staging server and installed via the
curl command. What I wanted to know was how I can update that to the latest
version and then push them changes to git for me to pull to live.

**Reply by justinbeaty** [reply]
Okay, I see. In this case I would probably just extract the latest magento-lts code on top of your existing repo.

**Reply by jparker1986** [reply]
Is there a simple way to do this with SSH? 

I understand the code is aimed at developers but a dummy guide for smaller store owners like myself maybe an idea?

**Reply by justinbeaty** [reply]
You could use this command, replacing `/path/to/magento/htdocs` with the path to your magento root.

```
curl -L https://github.com/OpenMage/magento-lts/archive/refs/tags/v19.4.15.tar.gz | tar xz --strip-components=1 -C /path/to/magento/htdocs
```

More docs could be helpful for sure.

**Comment by jparker1986** [comment]
Justin,

You're amazing, thank you - that was just the command I was looking for.

Out of interest I note that the latest version is is into the 20's now, how
would I find the URL link for the latest version?

On Fri, 18 Feb 2022 at 14:58, Justin Beaty ***@***.***> wrote:

> You could use this command, replacing /path/to/magento/htdocs with the
> path to your magento root.
>
> curl -L https://github.com/OpenMage/magento-lts/archive/refs/tags/v19.4.15.tar.gz | tar xz --strip-components=1 -C /path/to/magento/htdocs
>
> More docs could be helpful for sure.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenMage/magento-lts/discussions/2022#discussioncomment-2205317>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANMM4VPKYVYXN7767VPGWCTU3ZNAZANCNFSM5OSX2GQQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Jason Parker*

Parker Products Limited
Richmond Park
Richmond Road, Louth
Lincolnshire
LN11 0FU

t: 07872 379015
t: 01507 499198

**Reply by justinbeaty** [reply, marked as answer]
I linked to the 19.x release line just because you were upgrading an older store, but you can certainly go to the 20.x release. Note that both 19.x and 20.x are currently maintained, with the 20.x branch having some breaking changes.

To get the links, go to https://github.com/OpenMage/magento-lts/releases , find the release you want, go to "assets" and copy the "Source code (tar.gz)" link.

**Comment by jparker1986** [comment]
Brilliant,

I just tried the 20.x release and it broke my store front end.

I assume the 20.x release would require some in house dev to work?

Certainly feel a bit more of a "dummy guide" for upgrading would be useful.
The "migration" script is on the "Home Page" of OpenMage but not the
"Install" page which I found rather confusing.

On Fri, 18 Feb 2022 at 17:39, Justin Beaty ***@***.***> wrote:

> I linked to the 19.x release line just because you were upgrading an older
> store, but you can certainly go to the 20.x release. Note that both 19.x
> and 20.x are currently maintained, with the 20.x branch having some
> breaking changes.
>
> To get the links, go to https://github.com/OpenMage/magento-lts/releases
> , find the release you want, go to "assets" and copy the "Source code
> (tar.gz)" link.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenMage/magento-lts/discussions/2022#discussioncomment-2206407>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANMM4VMOCNSX4D6HMC4JBULU3Z73JANCNFSM5OSX2GQQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Jason Parker*

Parker Products Limited
Richmond Park
Richmond Road, Louth
Lincolnshire
LN11 0FU

t: 07872 379015
t: 01507 499198

**Reply by justinbeaty** [reply]
It probably depends on why 20.x broke your store, but of course make sure you cleared the cache after upgrading in case upgrade scripts need to run. Otherwise, you may be able to see why it broke in the exception log. There's no problem staying on the 19.x branch though, as all security updates and other optimizations get merged into that branch.

I think better docs might be something that could be addressed in the near future, but you're right most of us using OpenMage are developers I think.

**Comment by jparker1986** [comment]
Justin,

Your exactly right but it may help if more guidance was given on the actual
website about the project, what's new, what big fixes are done... Even some
screenshots showing backend updates.

Alot of stores are maybe not in a position to upgrade from M1 to M2 and if
I am one of those.

We have too many external systems developed around the M1 API and dB
layout.

Do any contributors provide outside work for integration and modifications?
If so where can I find them?
